src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -4586,6 +4586,74 @@ public class MyComponent : ComponentBase
         CompileToAssembly(generated);
     }
 
+    [IntegrationTestFact]
+    public void Component_WithStaticPageDirective()
+    {
+        // Act
+        var generated = CompileToCSharp(@"
+@page ""/MyPage""
+@staticpage
+@page ""/AnotherRoute/{id}""
+Hello");
+
+        // Assert
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        CompileToAssembly(generated);
+    }
+
+    [IntegrationTestFact]
+    public void Component_WithStaticPageDirective_DefinedInImportsFile()
+    {
+        // Arrange
+        var importItem = CreateProjectItem("_Imports.razor", "@staticpage", FileKinds.ComponentImport);
+        ImportItems.Add(importItem);
+
+        // Act
+        var generated = CompileToCSharp(@"
+@page ""/MyPage""
+@page ""/AnotherRoute/{id}""
+Hello
+");
+
+        // Assert
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        CompileToAssembly(generated);
+    }
+
+    [IntegrationTestFact]
+    public void Component_WithStaticPageDirective_ButNoPage()
+    {
+        var generated = CompileToCSharp(@"
+@staticpage
+Hello", throwOnFailure: false);
+
+        Assert.Collection(generated.Diagnostics, d =>
+        {
+            Assert.Equal("RZ10025", d.Id);
+            Assert.Equal(RazorDiagnosticSeverity.Error, d.Severity);
+            Assert.Equal("The @staticpage directive is only valid on components that also have a @page directive.", d.GetMessage(CultureInfo.InvariantCulture));
+        });
+    }
+
+    [IntegrationTestFact]
+    public void Component_WithStaticPageDirective_AndRenderMode()
+    {
+        var generated = CompileToCSharp(@"
+@page ""/url""
+@rendermode Microsoft.AspNetCore.Components.Web.RenderMode.Server
+@staticpage
```

Comment:
Consider adding test where `@staticpage` is defined somewhere inside the page, not in the header